mantidproject/mantid
# Use IArchiveSearch more selectively

Copied and rephrased from the issue:

The value of `datasearch.searcharchive` (case insensitive) will be interpreted differently in `FileFinder`:

* empty string or `off` means don't do archive search
* `on` means search your default facility only
* `all` means search in all facilities
* list of names means search facilities in the list (example is `hfir,sns` will search those and skip `isis`)

**To test:**

Turn logging up to `debug`, set your default facility to `isis` or `sns`, set your `datasearch.searcharchive` to various values, then search for `OSI12345` and `NOM12345`. I don't know if both exist, but you'll see from the logging whether or not it used archive search for the files.

Fixes #20352.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
